### PR TITLE
Notify amber users of negative credit immediately

### PR DESCRIPTION
**Pull request by wilco375**
Since amber users can now no longer order with negative credit at the start of an activity, it is nice if they are notified of their negative credit before the next activity, which is currently not always the case, since mails are always only sent on Thursday. This PR fixes this by sending an email immediately when a user moves from positive to negative credit

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -19,10 +19,18 @@ def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
     @order = Order.new(permitted_attributes.merge(created_by: current_user))
     authorize @order
 
+    current_credit = @order.user.credit
+
     if @order.save
+      if (@order.user.provider == 'amber_oauth2') && @order.user.credit.negative? && current_credit.positive?
```
Comment (DrumsnChocolate) [inline app/controllers/orders_controller.rb:25]: Will current_credit.positive not just point to the same value that @order.user.credit.positive points to? Am I missing something here?
Does @order.save have side effects?

**Review by DrumsnChocolate — COMMENTED** [review]
Will this also notify us in slack about how often notifications are sent about insuff credit, and to how many people? Since that is currently a thing

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (wilco375) [reply app/controllers/orders_controller.rb:25]: I wondered this too, but it retrieves the credit from the database again. Not sure if it always does this or just after saving (didn't test), but it does work in this context.

**Comment by wilco375** [comment]
@DrumsnChocolate no, that won't happen anymore then, also not for the treasurer. But I did not see a big benefit to this. Otherwise we'd have to store extra stuff in the database.

**Review by DrumsnChocolate — APPROVED** [review]
One last question: does this mean that amber users will only be notified once, or are reminders still sent to everyone once in a while?

**Comment by wilco375** [comment]
> One last question: does this mean that amber users will only be notified once, or are reminders still sent to everyone once in a while?

Only once, not every week anymore. But since one can't place any new orders when credit is negative this doesn't seem like a big problem to me (not that the reminders seemed to do much anyways before 😅 )

**Comment by DrumsnChocolate** [comment]
> not that the reminders seemed to do much anyways before :sweat_smile: 

fair enough lol

**Pull request merged**